### ISSUE TITLE
no-if rule doesn't work in many cases

### ISSUE THREAD

**Issue by CreativeTechGuy** [jest-community/eslint-plugin-jest#1026]
The `jest/no-if` rule has a bug which results in it failing to find conditions in many cases.

[This pops from the stack whenever a call expression ends](https://github.com/jest-community/eslint-plugin-jest/blob/main/src/rules/no-if.ts#L102-L104) yet [this only adds to the stack when specific call expression start](https://github.com/jest-community/eslint-plugin-jest/blob/main/src/rules/no-if.ts#L77-L85).

This results in more items being popped off than pushed.

Example:

```
test("shows error", () => {
    if (1 === 2) {
        expect(true).toBe(false);
    }
});

test("does not show error", () => {
    setTimeout(() => console.log("noop"));
    if (1 === 2) {
        expect(true).toBe(false);
    }
});
```

The inner function in the second example causes the stack to be popped off one too many times and so by the time it arrives at the condition, the stack is empty when it shouldn't be.

I believe there are also other cases where this can occur as when debugging I often see it attempting to pop off the stack several times despite the stack being empty.

**Comment by G-Rath** [comment]
Ironically I was looking at this code the other day when writing [this new rule (iirc)](https://github.com/jest-community/eslint-plugin-jest/pull/1013] and thinking "wait, isn't that check incorrect...?" 😅

**Comment by G-Rath** [comment]
I've "fixed" this by creating a new rule to replace `no-if`, which just doesn't allow any conditionals in tests.

**Comment by G-Rath** [comment]
This was closed in v26.1.0 but the bot didn't pick it up 😅

**Comment by SimenB** [comment]
it's due to rebase merge, so the bot lost track of the connection between the commit that landed in the PR that closed this issue and the one that landed on `main` (PR link in commit title is not enough)
